packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -325,19 +332,38 @@ export interface ClusterAttributes {
   readonly openIdConnectProvider?: iam.IOpenIdConnectProvider;
 
   /**
-   * An AWS Lambda Layer which includes `kubectl`, Helm and the AWS CLI. This layer
-   * is used by the kubectl handler to apply manifests and install helm charts.
+   * An AWS Lambda Layer which includes `kubectl` and Helm.
+   *
+   * This layer is used by the kubectl handler to apply manifests and install
+   * helm charts. You must pick an appropriate releases of one of the
+   * `@aws-cdk/asset-kubectl-vXX` packages, that works with the version of
+   * Kubernetes you have chosen. If you don't supply this value `kubectl`
+   * 1.20 will be used, but that version is most likely too old.
```

Comment:
Are the `@aws-cdk/asset-kubectl-vXX` packages ready for use already? If so, we should use them to pick the appropriate default based on the cluster version, if not, we shouldn't mention them here just yet.

Reply:
We can't, for multiple reasons:

- `@aws-cdk/asset-kubectl-vXX` will depend on `aws-cdk-lib` for the definition of `ILayerVersion`.
- We would need to depend on ALL possible versions of `@aws-cdk/asset-kubectl-vXX` so that we can pick at runtime, adding 5x20 MB to our download size. That's going against the goal of reducing size.

Reply:
Not sure I follow, wasn't the idea for them to be installed at synth time? no need to depend on them, just assume the appropriate package exists and install it when needed.

Reply:
We've come back from that:

- `npm install` at synth time: requires the user to have NPM, have a writeable `node_modules` directory, requires them to be non-network isolated, etc etc. Fraught with problems.
- Transparent loading if they have the package in their dep tree already: won't work for jsii languages. If the user did not explicitly interact with the dependency package in Python/Java/C#, the NPM module will not be extracted yet and our NPM code won't be able to load it.

Ultimate compromise: user installs the right package and references it in the API explicitly. Not *as* nice, but nice enough.

Reply:
Yeah I synced up with @madeline-k about this, wasn't aware of the backtrack. We did talk about a possible solution to the dynamic loading problem that will work across the board, but also agreed its not worth rocking the boat yet again at this point. The compromise you came up with seems reasonable, and we can improve on that later on if we feel its needed.